packages/react-native/Libraries/Components/TextInput/TextInput.js

```diff
@@ -1506,7 +1512,18 @@ function InternalTextInput(props: Props): React.Node {
     if (childCount > 1) {
       children = <Text>{children}</Text>;
     }
-
+    // For consistency with iOS set cursor/selectionHandle color as selectionColor
+    const colorProps = {
+      selectionColor: props.selectionColor,
+      selectionHandleColor:
+        props.selectionHandleColor === undefined
+          ? props.selectionColor
+          : props.selectionHandleColor,
+      cursorColor:
+        props.cursorColor === undefined
+          ? props.selectionColor
+          : props.cursorColor,
+    };
```

Comment:
Can you try define these props conditionally? We should only set the keys if they're actually being used.

```suggestion
    // Extract these at the top 
    const {selectionColor, selectionHandleColor, cursorColor} = otherProps;
    const colorProps = selectionColor != null ? {
        selectionColor,
        selectionHandleColor: selectionHandleColor === undefined
          ? selectionColor
          :  selectionHandleColor,
        cursorColor:
       cursorColor === undefined
          ?selectionColor
          :cursorColor,
    } : null;
```

Reply:
What do you think about this approach? That way, we'll allow setting `selectionHandleColor/cursorColor` even if `selectionColor` is not set
```suggestion
    const {selectionColor, selectionHandleColor, cursorColor} = otherProps;
    const colorProps = {
      selectionColor,
      selectionHandleColor: selectionHandleColor === undefined
          ? selectionColor
          : selectionHandleColor,
      cursorColor: cursorColor === undefined
          ? selectionColor
          : cursorColor,
    };
```

Reply:
I have updated the code to not duplicate these props in `colorProps` and `otherProps`, while I don't think there is an option to set these props conditionally and at the same time allow setting `selectionHandleColor/cursorColor` without setting `selectionColor`. What do you think about that?

Reply:
This seems reasonable. Alternatively you'd need to something like

```
let colorProps = null;
if (selectionColor != null || selectionHandleColor != null || cursorColor != nulll) {
```